### PR TITLE
feat: `psmLite` and `psm3` erc4626 arks

### PR DESCRIPTION
**Pull request by halaprix**
I'll help you create a comprehensive PR message based on the code changes. Let me analyze the changes and create a well-structured PR message.

## Description
This PR introduces two new Ark contracts that combine PSM (Peg Stability Module) functionality with ERC4626 vault integration. The contracts enable users to enter with USDC, swap to USDS/sUSDS, and optionally stake in vaults, providing flexible yield generation strategies.

## Changes
- Added `Psm3ERC4626Ark` contract that:
  - Supports direct USDS staking (USDC -> USDS -> ERC4626 vault)
  - Supports sUSDS staking (USDC -> sUSDS -> ERC4626 vault)
  - Integrates with PSM3 for USDC <-> sUSDS swaps
  - Implements ERC4626 vault deposits and withdrawals

- Added `PsmLiteERC4626Ark` contract that:
  - Supports direct USDS staking (USDC -> USDS -> ERC4626 vault)
  - Supports sUSDS staking (USDC -> USDS -> sUSDS -> ERC4626 vault)
  - Integrates with PSM Lite for USDC -> USDS swaps
  - Handles decimal conversion between 18 decimals and USDC decimals

- Added new interfaces:
  - `IPsm3.sol` for PSM3 contract interactions
  - `IPsmLite.sol` for PSM Lite contract interactions

## Benefits
1. Enhanced yield generation options for users through ERC4626 vault integration
2. Flexible staking strategies with support for both USDS and sUSDS
3. Improved capital efficiency by combining PSM swaps with vault staking
4. Standardized interface for PSM interactions through dedicated interfaces
5. Robust error handling and input validation for all operations

## Testing
The changes should be tested through:
1. Unit tests for both Ark contracts covering:
   - Boarding and disembarking flows
   - USDS and sUSDS staking scenarios
   - Decimal conversion handling
   - Error cases and edge conditions
2. Integration tests with:
   - PSM3 and PSM Lite contracts
   - ERC4626 vault implementations
   - USDS and sUSDS token contracts
3. Fuzz testing for amount calculations and conversions

## Next steps
- [ ] Add comprehensive test coverage for both Ark contracts
- [ ] Add documentation for integration with different vault types
- [ ] Consider adding events for important state changes
- [ ] Review gas optimization opportunities
- [ ] Consider adding emergency pause functionality

## Additional Notes
- The contracts use OpenZeppelin's SafeERC20 for secure token transfers
- Both contracts implement the base Ark interface and inherit its functionality
- The contracts handle decimal conversions appropriately, especially in PsmLiteERC4626Ark
- No harvest function is needed as rewards are automatically compounded in sUSDS

Please review and provide any feedback or suggestions for improvement.

## Release Notes

* **New Features**
  * Added two new vault integration types: PSM3 ERC4626 Ark and PSM Lite ERC4626 Ark, enabling users to deposit USDC into ERC4626 vaults with optional staking in derivative tokens.
  * New vault options now available on mainnet and Base networks.

* **Tests**
  * Added comprehensive test coverage for new vault integration types.

* **Chores**
  * Updated deployment configurations and infrastructure to support new vault types.

### CODE REVIEW

File: packages/core-contracts/src/contracts/arks/Psm3ERC4626Ark.sol [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        uint256 tokenAmount = psm.swapExactIn(
+            address(config.asset),
+            address(shouldStake ? susds : usds),
+            amount,
+            expectedTokenAmount,
+            address(this),
+            0
+        );
+
+        if (shouldStake) {
```
Comment (robercano) [inline packages/core-contracts/src/contracts/arks/Psm3ERC4626Ark.sol]: Opinion: we could reverse this and use the `_underlying` function mentioned before to fetch the asset as an IERC20 and then pass it to `_handleBoarding` that uses the IERC20 interface to do the boarding.

**Review by robercano — APPROVED** [review]
Just a minor comment and some nice-to-have based on my personal opinion. Approved anyway

**Review by robercano — APPROVED** [review]

File: packages/core-contracts/src/contracts/arks/Psm3ERC4626Ark.sol [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    /*//////////////////////////////////////////////////////////////
+                            PUBLIC FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+    /**
+     * @notice Returns the total assets in the vault
+     * @return assets Total assets in USDC terms
+     */
+    function totalAssets() public view override returns (uint256 assets) {
+        uint256 balance = erc4626Vault.balanceOf(address(this));
+        if (balance > 0) {
```
Comment (robercano) [inline packages/core-contracts/src/contracts/arks/Psm3ERC4626Ark.sol]: Is this the usual case? If in general we expect the vault to have a non-zero balance, this is actually hurting the gas optimization (as 99.99% of the time the balance we are not zero but we are still checking it)

File: packages/core-contracts/src/contracts/arks/PsmLiteERC4626Ark.sol [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+     * @return withdrawableAssets Amount of assets that can be withdrawn
+     */
+    function _withdrawableTotalAssets()
+        internal
+        view
+        override
+        returns (uint256 withdrawableAssets)
+    {
+        uint256 shares = erc4626Vault.balanceOf(address(this));
+        if (shares > 0) {
```
Comment (robercano) [inline packages/core-contracts/src/contracts/arks/PsmLiteERC4626Ark.sol]: And here

**Review by robercano — CHANGES_REQUESTED** [review]

**Review by robercano — APPROVED** [review]
